### ISSUE TITLE
Multiple Stores Support

### ISSUE THREAD

**Issue by adrum** [medusajs/medusa#384]
On the flip side of #383, I am interested in tracking the progress of this idea:

>On Multi-tenant functionality: This feature will come soon; we will launch a concept called "Stores" in the near future to enable you to control multiple shops/branches from one place

I imagine this would be a single business/entity with multiple physical locations. This would allow the different branches to maintain their own inventory levels. Customers would be able to use one account across all locations/branches. However, they would only be actively shopping one branch at a time, and would allow them to switch between branches to find the one physically closest to them. In the event of a product being out of stock, this would also allow customers to find a location near them with stock (or at least switch locations, and then manually check).

From the admin perspective, you would be able to see aggregated stats across all branches.

Is this what you had in mind?

**Comment by klauss194** [comment]
+1

**Comment by LucasCioffi** [comment]
@adrum @klauss194 have you heard any news about multi-tenant functionality on Medusa?  Thanks for raising the question last year.

**Comment by err0x1a** [comment]
It would be great to have multi-tenant support. I'm working on a project where independent clients will have their own online stores, and I considered using MedusaJS solely as an API for both a custom admin and a custom storefront. However, I realized that I wouldn't be able to have a separate store for each client on the same server.

**Comment by olivermrbl** [comment]
@adrum – sorry for the slow response. 

We've recently announced a Multi-warehouse API allowing you to define several stock locations for product variants. I realize it is not exactly what you are requesting, but I recommend giving [the documentation](https://docs.medusajs.com/modules/multiwarehouse/overview) a read.

Would love to know what you think, and if could meet your requirements.

Closing this, as we will not be adding support for multiple Stores.
